components/brave_wallet_ui/common/async/lib.ts

```diff
@@ -501,10 +501,19 @@ export function refreshFullNetworkList () {
       return networkList.networks
     }))
     const flattenedNetworkList = getFullNetworkList.flat(1)
-    const networkList =
+    let networkList =
       isTestNetworksEnabled.isEnabled
         ? flattenedNetworkList
         : flattenedNetworkList.filter((network) => !SupportedTestNetworks.includes(network.chainId))
+
+    const defaultEthChainId = (await jsonRpcService.getChainId(BraveWallet.CoinType.ETH)).chainId
+    const hiddenEthNetworkList = (await jsonRpcService.getHiddenNetworks(BraveWallet.CoinType.ETH)).chainIds
+    networkList = networkList.filter((network: BraveWallet.NetworkInfo) => {
+      return !(network.coin === BraveWallet.CoinType.ETH &&
+        network.chainId !== defaultEthChainId &&
```

Comment:
Instead of doing this, maybe we should reset the network to the mainnet of that coin type when users try to hide the selected(active) network? Otherwise it's confusing that I see the icon is changed to hidden but can still see the network.
The same thing can be applied to `RemoveCustomNetwork` too, which is https://github.com/brave/brave-browser/issues/21344.

Reply:
Discussed to hide "eye" button for active network